src/stdlib/val_operation.rs

```diff
@@ -136,16 +137,39 @@ fndr_native_func!(
         match (a.unwrap_value(), b.unwrap_value()) {
             (String(a), String(b)) => Ok(FenderValue::make_string(format!("{}{}", *a, *b)).into()),
             (List(a), List(b)) => {
-                let mut newl = a.to_vec();
-                newl.extend(b.iter().cloned());
-                Ok(FenderValue::make_list(newl).into())
+                let mut new_list = a.to_vec();
+                new_list.extend(b.iter().cloned());
+                Ok(FenderValue::make_list(new_list).into())
             }
             _ => Ok(FenderValue::make_error(format!(
-                "Cannot concat {} with {}",
+                "Cannot call concat on type `{}` and `{}`",
                 a.get_real_type_id().to_string(),
                 b.get_real_type_id().to_string()
             ))
             .into()),
         }
     }
 );
+
+fndr_native_func!(
+    /// Insert a key-value pair into a HashMap, or insert into a list at a given index
+    insert_func,
+    |_, mut collection, key, value| {
+        let res = match collection.get_real_type_id() {
+            FenderTypeId::HashMap => (*collection).insert(key.unwrap_value(), value),
+            FenderTypeId::List if key.get_real_type_id() == FenderTypeId::Int => {
+                collection.insert(key.unwrap_value(), value)
+            }
+            FenderTypeId::List => Err(format!(
+                "Cannot index `List` with type `{}`",
+                key.get_type_id().to_string()
+            )),
```

Comment:
Could this insert substrings into strings too?

Reply:
that and the same for swap can be added later because they are more involved for strings
this is just to get the minimum functionality for HashMaps in there 
I'll make an issue for the string implementations